### PR TITLE
actions: should be able to make use of the make commands.

### PR DESCRIPTION
**Pull request by jimmylee**
# GitHub Actions

This commit will add a GitHub action that will automatically:

+ Check out the repo
+ Build the repo
+ Publish the repo

Whenever master is updated.

### CODE REVIEW

**Comment by jimmylee** [comment]
Running into an issue:

```
##[error]/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-core@v0.3.0/crypto/ed25519.go:5:2: unknown import path "crypto/ed25519": cannot find module providing package crypto/ed25519
##[error]Makefile:237: recipe for target 'build-code' failed
```

Will ping @Stebalien and see if he knows of a way to work around this.

**Comment by sternhenri** [comment]
I've seen the above issue before, I believe it is related to your go version. I believe you may be running on `1.12` rather than `1.13` in your deployment. Check out https://golang.org/doc/go1.13

**Comment by jimmylee** [comment]
@sternhenri that is correct, and @Stebalien confirmed this last night via DM.

However in my tests, `1.13.8` is being installed, `go version` is verifying that version exists, but the `make *` is still using the older version and I'm trying to figure out whats going on there.

**Comment by jimmylee** [comment]
Screen Shot 2020-02-13 at 12 31 37 PM

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    steps:
+      - uses: actions/checkout@v1
+        with:
+          submodules: true
+
+      - name: Set GOPATH
+        run: |
+          echo "::set-env name=GOPATH::${{ github.workspace }}/go"
+          echo "::add-path::${{ github.workspace }}/go/bin"
+        shell: bash
```
Comment (Kubuxu) [inline .github/workflows/main.yml]: I would remove this step and retry.

Comment (jimmylee) [reply .github/workflows/main.yml]: Removed. But did not resolve the issue.

**Comment by jimmylee** [comment]
Made progress! New error:

```
Building sites … ERROR 2020/02/14 09:56:09 Transformation failed: TOCSS: failed to transform "book.scss" (text/x-scss): this feature is not available in your current Hugo version, see https://goo.gl/YMrWcn for more information
WARN 2020/02/14 09:56:09 Page.Dir is deprecated and will be removed in a future release. Use .File.Dir
WARN 2020/02/14 09:56:09 Bad include org.Keyword{Key:"INCLUDE", Value:"\"../../build/orient/filecoin.json\" src json"}: file does not exist
Error: Error building site: logged 1 error(s)
Total in 1704 ms
```

**Pull request closed without merging**